cmd/iam.go

```diff
@@ -1355,11 +1336,12 @@ func (sys *IAMSys) GetUser(accessKey string) (cred auth.Credentials, ok bool) {
 		return cred, false
 	}
 
-	sys.store.rlock()
-	fallback := sys.storeFallback
-	sys.store.runlock()
-	if fallback {
+	fallback := false
+	select {
+	case <-sys.configLoaded:
+	default:
 		sys.loadUserFromStore(accessKey)
+		fallback = true
```

Comment:
I am not very familliar with channels but since this is a default behaviour, is there any chance for fallback to be false in the next line of this method ?
because there is a test on the fallback value line 1349, and that puzzles me.

Reply:
This is a non-blocking select call. If the config has been loaded the `case <-sys.configLoaded:` will be selected if the channel is blocked, in this case not closed. This is in contrast to `<-sys.configLoaded` which will block until the channel is closed.

This means that when `close(sys.configLoaded)` above has been called fallback will be false, since we have completed the initial load.

Reply:
ok thanks, in fact I was too used to the Java switch which water falls to the next statement, here you have 2 distincts statement and that is clear.